<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><head></head><body>
































































<div class="container-fluid main-container">




<div>

<div class="btn-group pull-right float-right">
<span>Code</span> <span class="caret"></span>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a rel="noopener" href="#">Show All Code</a></li>
<li><a rel="noopener" href="#">Hide All Code</a></li>
<li class="divider"></li>
<li><a rel="noopener" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RSEM Raw Data DESEQ2 Neurotoxicity Study</h1>

</div>



<p>The raw data for this study was kindly provided by the authors of the paper:</p>
<p>Schwartz, M. P., Hou, Z., Propson, N. E., Zhang, J., Engstrom, C. J., Costa, V. S., Jiang, P., Nguyen, B. K., Bolin, J. M., Daly, W., Wang, Y., Stewart, R., Page, C. D., Murphy, W. L., &amp; Thomson, J. A. (2015). Human pluripotent stem cell-derived neural constructs for predicting neural toxicity. Proceedings of the National Academy of Sciences, 112(40), 12516–12521. <a rel="noopener" href="https://doi.org/10.1073/pnas.1516645112" class="uri">https://doi.org/10.1073/pnas.1516645112</a></p>
<p>The data is Expected Count data processed with RSEM.</p>



<pre class="r"><code>install.packages(&#39;dendextend&#39;)</code></pre>


<pre><code>trying URL &#39;https://cran.rstudio.com/bin/macosx/contrib/4.1/dendextend_1.15.2.tgz&#39;
Content type &#39;application/x-gzip&#39; length 3885614 bytes (3.7 MB)
==================================================
downloaded 3.7 MB</code></pre>


<pre><code>
The downloaded binary packages are in
    /var/folders/kz/671nytjx6wbb6qvvfpv7jwt00000gn/T//RtmpARyGGd/downloaded_packages</code></pre>




<pre class="r"><code>install.packages(&quot;circlize&quot;)</code></pre>


<pre><code>also installing the dependencies ‘GlobalOptions’, ‘shape’

trying URL &#39;https://cran.rstudio.com/bin/macosx/contrib/4.1/GlobalOptions_0.1.2.tgz&#39;
Content type &#39;application/x-gzip&#39; length 459466 bytes (448 KB)
==================================================
downloaded 448 KB

trying URL &#39;https://cran.rstudio.com/bin/macosx/contrib/4.1/shape_1.4.6.tgz&#39;
Content type &#39;application/x-gzip&#39; length 785275 bytes (766 KB)
==================================================
downloaded 766 KB

trying URL &#39;https://cran.rstudio.com/bin/macosx/contrib/4.1/circlize_0.4.14.tgz&#39;
Content type &#39;application/x-gzip&#39; length 3307025 bytes (3.2 MB)
==================================================
downloaded 3.2 MB</code></pre>


<pre><code>
The downloaded binary packages are in
    /var/folders/kz/671nytjx6wbb6qvvfpv7jwt00000gn/T//RtmpARyGGd/downloaded_packages</code></pre>






<pre class="r"><code>library(dplyr)
library(pheatmap)
library(DESeq2)</code></pre>




<pre class="r"><code>library(dendextend)</code></pre>




<pre class="r"><code>library(ggplot2)</code></pre>






<pre class="r"><code>RSEM_data=read.csv(&#39;/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/RSEM_data.csv&#39;)</code></pre>



<p>The RSEM data is made up of the data 2 and 7 days after chemical exposure.</p>



<pre class="r"><code>head(RSEM_data)</code></pre>


<div>

</div>




<pre class="r"><code>RSEM_day2&lt;-RSEM_data[,1:146]</code></pre>




<pre class="r"><code>RSEM_day2</code></pre>


<div>

</div>






<pre class="r"><code>#Obtain data from day 2
library(plyr)
RSEM_day2 &lt;- plyr::rename(RSEM_day2, c(&quot;d2t11a_reprep&quot;=&quot;d2t11a&quot;))</code></pre>






<pre class="r"><code>RSEM_day2 &lt;- plyr::rename(RSEM_day2, c(&quot;d2t13a_reprep&quot;=&quot;d2t13a&quot;))</code></pre>






<pre class="r"><code>RSEM_day2&lt;-RSEM_day2[,order(colnames(RSEM_day2))]
RSEM_day2$Transcript_ID&lt;-NULL
RSEM_day2</code></pre>


<div>

</div>






<pre class="r"><code>#Obtain data from day 7
RSEM_day7&lt;-RSEM_data[,147:290]
RSEM_day7&lt;-RSEM_day7[,order(colnames(RSEM_day7))]
RSEM_day7[&#39;Gene.ID&#39;]&lt;-RSEM_day2$Gene.ID
head(RSEM_day7)</code></pre>


<div>

</div>


<pre class="r"><code>NA
NA</code></pre>



<p>Making sure the columns correspond to one another:</p>



<pre class="r"><code>colnames(RSEM_day7)</code></pre>


<pre><code>  [1] &quot;d7b10a&quot;  &quot;d7b10b&quot;  &quot;d7b1a&quot;   &quot;d7b1b&quot;   &quot;d7b2a&quot;   &quot;d7b2b&quot;   &quot;d7b3a&quot;  
  [8] &quot;d7b3b&quot;   &quot;d7b4a&quot;   &quot;d7b4b&quot;   &quot;d7b5a&quot;   &quot;d7b5b&quot;   &quot;d7b6a&quot;   &quot;d7b6b&quot;  
 [15] &quot;d7b7a&quot;   &quot;d7b7b&quot;   &quot;d7b8a&quot;   &quot;d7b8b&quot;   &quot;d7b9a&quot;   &quot;d7b9b&quot;   &quot;d7c10a&quot; 
 [22] &quot;d7c10b&quot;  &quot;d7c11a&quot;  &quot;d7c11b&quot;  &quot;d7c12a&quot;  &quot;d7c12b&quot;  &quot;d7c13a&quot;  &quot;d7c13b&quot; 
 [29] &quot;d7c14a&quot;  &quot;d7c14b&quot;  &quot;d7c15a&quot;  &quot;d7c15b&quot;  &quot;d7c16a&quot;  &quot;d7c16b&quot;  &quot;d7c17a&quot; 
 [36] &quot;d7c17b&quot;  &quot;d7c18a&quot;  &quot;d7c18b&quot;  &quot;d7c19a&quot;  &quot;d7c19b&quot;  &quot;d7c1a&quot;   &quot;d7c1b&quot;  
 [43] &quot;d7c20a&quot;  &quot;d7c20b&quot;  &quot;d7c21a&quot;  &quot;d7c21b&quot;  &quot;d7c22a&quot;  &quot;d7c22b&quot;  &quot;d7c23a&quot; 
 [50] &quot;d7c23b&quot;  &quot;d7c24a&quot;  &quot;d7c24b&quot;  &quot;d7c25a&quot;  &quot;d7c25b&quot;  &quot;d7c26a&quot;  &quot;d7c26b&quot; 
 [57] &quot;d7c2a&quot;   &quot;d7c2b&quot;   &quot;d7c3a&quot;   &quot;d7c3b&quot;   &quot;d7c4a&quot;   &quot;d7c4b&quot;   &quot;d7c5a&quot;  
 [64] &quot;d7c5b&quot;   &quot;d7c6a&quot;   &quot;d7c6b&quot;   &quot;d7c7a&quot;   &quot;d7c7b&quot;   &quot;d7c8a&quot;   &quot;d7c8b&quot;  
 [71] &quot;d7c9a&quot;   &quot;d7c9b&quot;   &quot;d7t10a&quot;  &quot;d7t10b&quot;  &quot;d7t11a&quot;  &quot;d7t11b&quot;  &quot;d7t12a&quot; 
 [78] &quot;d7t12b&quot;  &quot;d7t13a&quot;  &quot;d7t13b&quot;  &quot;d7t14a&quot;  &quot;d7t14b&quot;  &quot;d7t15a&quot;  &quot;d7t15b&quot; 
 [85] &quot;d7t16a&quot;  &quot;d7t16b&quot;  &quot;d7t17a&quot;  &quot;d7t17b&quot;  &quot;d7t18a&quot;  &quot;d7t18b&quot;  &quot;d7t19a&quot; 
 [92] &quot;d7t19b&quot;  &quot;d7t1a&quot;   &quot;d7t1b&quot;   &quot;d7t20a&quot;  &quot;d7t20b&quot;  &quot;d7t21a&quot;  &quot;d7t21b&quot; 
 [99] &quot;d7t22a&quot;  &quot;d7t22b&quot;  &quot;d7t23a&quot;  &quot;d7t23b&quot;  &quot;d7t24a&quot;  &quot;d7t24b&quot;  &quot;d7t25a&quot; 
[106] &quot;d7t25b&quot;  &quot;d7t26a&quot;  &quot;d7t26b&quot;  &quot;d7t27a&quot;  &quot;d7t27b&quot;  &quot;d7t28a&quot;  &quot;d7t28b&quot; 
[113] &quot;d7t29a&quot;  &quot;d7t29b&quot;  &quot;d7t2a&quot;   &quot;d7t2b&quot;   &quot;d7t30a&quot;  &quot;d7t30b&quot;  &quot;d7t32a&quot; 
[120] &quot;d7t32b&quot;  &quot;d7t33a&quot;  &quot;d7t33b&quot;  &quot;d7t34a&quot;  &quot;d7t34b&quot;  &quot;d7t35a&quot;  &quot;d7t35b&quot; 
[127] &quot;d7t3a&quot;   &quot;d7t3b&quot;   &quot;d7t4a&quot;   &quot;d7t4b&quot;   &quot;d7t5a&quot;   &quot;d7t5b&quot;   &quot;d7t6a&quot;  
[134] &quot;d7t6b&quot;   &quot;d7t7a&quot;   &quot;d7t7b&quot;   &quot;d7t8a&quot;   &quot;d7t8b&quot;   &quot;d7t9a&quot;   &quot;d7t9b&quot;  
[141] &quot;d7un1a&quot;  &quot;d7un1b&quot;  &quot;d7un2a&quot;  &quot;d7un2b&quot;  &quot;Gene.ID&quot;</code></pre>




<pre class="r"><code>colnames(RSEM_day2)</code></pre>


<pre><code>  [1] &quot;d2b10a&quot;  &quot;d2b10b&quot;  &quot;d2b1a&quot;   &quot;d2b1b&quot;   &quot;d2b2a&quot;   &quot;d2b2b&quot;   &quot;d2b3a&quot;  
  [8] &quot;d2b3b&quot;   &quot;d2b4a&quot;   &quot;d2b4b&quot;   &quot;d2b5a&quot;   &quot;d2b5b&quot;   &quot;d2b6a&quot;   &quot;d2b6b&quot;  
 [15] &quot;d2b7a&quot;   &quot;d2b7b&quot;   &quot;d2b8a&quot;   &quot;d2b8b&quot;   &quot;d2b9a&quot;   &quot;d2b9b&quot;   &quot;d2c10a&quot; 
 [22] &quot;d2c10b&quot;  &quot;d2c11a&quot;  &quot;d2c11b&quot;  &quot;d2c12a&quot;  &quot;d2c12b&quot;  &quot;d2c13a&quot;  &quot;d2c13b&quot; 
 [29] &quot;d2c14a&quot;  &quot;d2c14b&quot;  &quot;d2c15a&quot;  &quot;d2c15b&quot;  &quot;d2c16a&quot;  &quot;d2c16b&quot;  &quot;d2c17a&quot; 
 [36] &quot;d2c17b&quot;  &quot;d2c18a&quot;  &quot;d2c18b&quot;  &quot;d2c19a&quot;  &quot;d2c19b&quot;  &quot;d2c1a&quot;   &quot;d2c1b&quot;  
 [43] &quot;d2c20a&quot;  &quot;d2c20b&quot;  &quot;d2c21a&quot;  &quot;d2c21b&quot;  &quot;d2c22a&quot;  &quot;d2c22b&quot;  &quot;d2c23a&quot; 
 [50] &quot;d2c23b&quot;  &quot;d2c24a&quot;  &quot;d2c24b&quot;  &quot;d2c25a&quot;  &quot;d2c25b&quot;  &quot;d2c26a&quot;  &quot;d2c26b&quot; 
 [57] &quot;d2c2a&quot;   &quot;d2c2b&quot;   &quot;d2c3a&quot;   &quot;d2c3b&quot;   &quot;d2c4a&quot;   &quot;d2c4b&quot;   &quot;d2c5a&quot;  
 [64] &quot;d2c5b&quot;   &quot;d2c6a&quot;   &quot;d2c6b&quot;   &quot;d2c7a&quot;   &quot;d2c7b&quot;   &quot;d2c8a&quot;   &quot;d2c8b&quot;  
 [71] &quot;d2c9a&quot;   &quot;d2c9b&quot;   &quot;d2t10a&quot;  &quot;d2t10b&quot;  &quot;d2t11a&quot;  &quot;d2t11b&quot;  &quot;d2t12a&quot; 
 [78] &quot;d2t12b&quot;  &quot;d2t13a&quot;  &quot;d2t13b&quot;  &quot;d2t14a&quot;  &quot;d2t14b&quot;  &quot;d2t15a&quot;  &quot;d2t15b&quot; 
 [85] &quot;d2t16a&quot;  &quot;d2t16b&quot;  &quot;d2t17a&quot;  &quot;d2t17b&quot;  &quot;d2t18a&quot;  &quot;d2t18b&quot;  &quot;d2t19a&quot; 
 [92] &quot;d2t19b&quot;  &quot;d2t1a&quot;   &quot;d2t1b&quot;   &quot;d2t20a&quot;  &quot;d2t20b&quot;  &quot;d2t21a&quot;  &quot;d2t21b&quot; 
 [99] &quot;d2t22a&quot;  &quot;d2t22b&quot;  &quot;d2t23a&quot;  &quot;d2t23b&quot;  &quot;d2t24a&quot;  &quot;d2t24b&quot;  &quot;d2t25a&quot; 
[106] &quot;d2t25b&quot;  &quot;d2t26a&quot;  &quot;d2t26b&quot;  &quot;d2t27a&quot;  &quot;d2t27b&quot;  &quot;d2t28a&quot;  &quot;d2t28b&quot; 
[113] &quot;d2t29a&quot;  &quot;d2t29b&quot;  &quot;d2t2a&quot;   &quot;d2t2b&quot;   &quot;d2t30a&quot;  &quot;d2t30b&quot;  &quot;d2t32a&quot; 
[120] &quot;d2t32b&quot;  &quot;d2t33a&quot;  &quot;d2t33b&quot;  &quot;d2t34a&quot;  &quot;d2t34b&quot;  &quot;d2t35a&quot;  &quot;d2t35b&quot; 
[127] &quot;d2t3a&quot;   &quot;d2t3b&quot;   &quot;d2t4a&quot;   &quot;d2t4b&quot;   &quot;d2t5a&quot;   &quot;d2t5b&quot;   &quot;d2t6a&quot;  
[134] &quot;d2t6b&quot;   &quot;d2t7a&quot;   &quot;d2t7b&quot;   &quot;d2t8a&quot;   &quot;d2t8b&quot;   &quot;d2t9a&quot;   &quot;d2t9b&quot;  
[141] &quot;d2un1a&quot;  &quot;d2un1b&quot;  &quot;d2un2a&quot;  &quot;d2un2b&quot;  &quot;Gene.ID&quot;</code></pre>



<p>The datafrmes are now sorted and now they can be added in order to obtain a dataframe that represents the average gene expression from day 2 and day 7:</p>



<pre class="r"><code>RSEM_avg&lt;-(RSEM_day2[,1:144]+RSEM_day7[,1:144])/2</code></pre>



<p>The dataframe below can now be used for exploratory data analysis and differential gene expression analysis.</p>



<pre class="r"><code>head(RSEM_avg)</code></pre>


<div>

</div>




<pre class="r"><code>#Remove last four columns labeled &#39;un&#39;
RSEM_avg$d2un1a=NULL
RSEM_avg$d2un1b=NULL
RSEM_avg$d2un2a=NULL
RSEM_avg$d2un2b=NULL</code></pre>






<pre class="r"><code>#set rownames to gene id
rownames(RSEM_avg)&lt;-RSEM_data$Gene.ID</code></pre>






<pre class="r"><code>head(RSEM_avg)</code></pre>


<div>

</div>



<p>Save Dataframe to csv file:</p>



<pre class="r"><code>write.csv(RSEM_avg,&#39;/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/RSEM_avg.csv&#39;, row.names = FALSE)</code></pre>



<div class="section level3">
<h3>Differential Gene Expression Analysis</h3>



<pre class="r"><code>expgroup&lt;-data.frame()
condition_toxic&lt;-rep(&#39;Toxic&#39;,68,)
condition_nontoxic&lt;-rep(&#39;Nontoxic&#39;,72)
condition&lt;-c(condition_nontoxic,condition_toxic)
samples&lt;-colnames(RSEM_avg)
expgroup&lt;-cbind(condition)
rownames(expgroup)&lt;-samples</code></pre>






<pre class="r"><code>expgroup</code></pre>


<pre><code>       condition 
d2b10a &quot;Nontoxic&quot;
d2b10b &quot;Nontoxic&quot;
d2b1a  &quot;Nontoxic&quot;
d2b1b  &quot;Nontoxic&quot;
d2b2a  &quot;Nontoxic&quot;
d2b2b  &quot;Nontoxic&quot;
d2b3a  &quot;Nontoxic&quot;
d2b3b  &quot;Nontoxic&quot;
d2b4a  &quot;Nontoxic&quot;
d2b4b  &quot;Nontoxic&quot;
d2b5a  &quot;Nontoxic&quot;
d2b5b  &quot;Nontoxic&quot;
d2b6a  &quot;Nontoxic&quot;
d2b6b  &quot;Nontoxic&quot;
d2b7a  &quot;Nontoxic&quot;
d2b7b  &quot;Nontoxic&quot;
d2b8a  &quot;Nontoxic&quot;
d2b8b  &quot;Nontoxic&quot;
d2b9a  &quot;Nontoxic&quot;
d2b9b  &quot;Nontoxic&quot;
d2c10a &quot;Nontoxic&quot;
d2c10b &quot;Nontoxic&quot;
d2c11a &quot;Nontoxic&quot;
d2c11b &quot;Nontoxic&quot;
d2c12a &quot;Nontoxic&quot;
d2c12b &quot;Nontoxic&quot;
d2c13a &quot;Nontoxic&quot;
d2c13b &quot;Nontoxic&quot;
d2c14a &quot;Nontoxic&quot;
d2c14b &quot;Nontoxic&quot;
d2c15a &quot;Nontoxic&quot;
d2c15b &quot;Nontoxic&quot;
d2c16a &quot;Nontoxic&quot;
d2c16b &quot;Nontoxic&quot;
d2c17a &quot;Nontoxic&quot;
d2c17b &quot;Nontoxic&quot;
d2c18a &quot;Nontoxic&quot;
d2c18b &quot;Nontoxic&quot;
d2c19a &quot;Nontoxic&quot;
d2c19b &quot;Nontoxic&quot;
d2c1a  &quot;Nontoxic&quot;
d2c1b  &quot;Nontoxic&quot;
d2c20a &quot;Nontoxic&quot;
d2c20b &quot;Nontoxic&quot;
d2c21a &quot;Nontoxic&quot;
d2c21b &quot;Nontoxic&quot;
d2c22a &quot;Nontoxic&quot;
d2c22b &quot;Nontoxic&quot;
d2c23a &quot;Nontoxic&quot;
d2c23b &quot;Nontoxic&quot;
d2c24a &quot;Nontoxic&quot;
d2c24b &quot;Nontoxic&quot;
d2c25a &quot;Nontoxic&quot;
d2c25b &quot;Nontoxic&quot;
d2c26a &quot;Nontoxic&quot;
d2c26b &quot;Nontoxic&quot;
d2c2a  &quot;Nontoxic&quot;
d2c2b  &quot;Nontoxic&quot;
d2c3a  &quot;Nontoxic&quot;
d2c3b  &quot;Nontoxic&quot;
d2c4a  &quot;Nontoxic&quot;
d2c4b  &quot;Nontoxic&quot;
d2c5a  &quot;Nontoxic&quot;
d2c5b  &quot;Nontoxic&quot;
d2c6a  &quot;Nontoxic&quot;
d2c6b  &quot;Nontoxic&quot;
d2c7a  &quot;Nontoxic&quot;
d2c7b  &quot;Nontoxic&quot;
d2c8a  &quot;Nontoxic&quot;
d2c8b  &quot;Nontoxic&quot;
d2c9a  &quot;Nontoxic&quot;
d2c9b  &quot;Nontoxic&quot;
d2t10a &quot;Toxic&quot;   
d2t10b &quot;Toxic&quot;   
d2t11a &quot;Toxic&quot;   
d2t11b &quot;Toxic&quot;   
d2t12a &quot;Toxic&quot;   
d2t12b &quot;Toxic&quot;   
d2t13a &quot;Toxic&quot;   
d2t13b &quot;Toxic&quot;   
d2t14a &quot;Toxic&quot;   
d2t14b &quot;Toxic&quot;   
d2t15a &quot;Toxic&quot;   
d2t15b &quot;Toxic&quot;   
d2t16a &quot;Toxic&quot;   
d2t16b &quot;Toxic&quot;   
d2t17a &quot;Toxic&quot;   
d2t17b &quot;Toxic&quot;   
d2t18a &quot;Toxic&quot;   
d2t18b &quot;Toxic&quot;   
d2t19a &quot;Toxic&quot;   
d2t19b &quot;Toxic&quot;   
d2t1a  &quot;Toxic&quot;   
d2t1b  &quot;Toxic&quot;   
d2t20a &quot;Toxic&quot;   
d2t20b &quot;Toxic&quot;   
d2t21a &quot;Toxic&quot;   
d2t21b &quot;Toxic&quot;   
d2t22a &quot;Toxic&quot;   
d2t22b &quot;Toxic&quot;   
d2t23a &quot;Toxic&quot;   
d2t23b &quot;Toxic&quot;   
d2t24a &quot;Toxic&quot;   
d2t24b &quot;Toxic&quot;   
d2t25a &quot;Toxic&quot;   
d2t25b &quot;Toxic&quot;   
d2t26a &quot;Toxic&quot;   
d2t26b &quot;Toxic&quot;   
d2t27a &quot;Toxic&quot;   
d2t27b &quot;Toxic&quot;   
d2t28a &quot;Toxic&quot;   
d2t28b &quot;Toxic&quot;   
d2t29a &quot;Toxic&quot;   
d2t29b &quot;Toxic&quot;   
d2t2a  &quot;Toxic&quot;   
d2t2b  &quot;Toxic&quot;   
d2t30a &quot;Toxic&quot;   
d2t30b &quot;Toxic&quot;   
d2t32a &quot;Toxic&quot;   
d2t32b &quot;Toxic&quot;   
d2t33a &quot;Toxic&quot;   
d2t33b &quot;Toxic&quot;   
d2t34a &quot;Toxic&quot;   
d2t34b &quot;Toxic&quot;   
d2t35a &quot;Toxic&quot;   
d2t35b &quot;Toxic&quot;   
d2t3a  &quot;Toxic&quot;   
d2t3b  &quot;Toxic&quot;   
d2t4a  &quot;Toxic&quot;   
d2t4b  &quot;Toxic&quot;   
d2t5a  &quot;Toxic&quot;   
d2t5b  &quot;Toxic&quot;   
d2t6a  &quot;Toxic&quot;   
d2t6b  &quot;Toxic&quot;   
d2t7a  &quot;Toxic&quot;   
d2t7b  &quot;Toxic&quot;   
d2t8a  &quot;Toxic&quot;   
d2t8b  &quot;Toxic&quot;   
d2t9a  &quot;Toxic&quot;   
d2t9b  &quot;Toxic&quot;   </code></pre>




<pre class="r"><code>#Round values in count matrix for DESEQ2 input
RSEM_avg&lt;-round(RSEM_avg,0)</code></pre>






<pre class="r"><code>dds &lt;-DESeqDataSetFromMatrix(
  countData=RSEM_avg,
  colData=expgroup,
  design=~condition,
)</code></pre>


<pre><code>converting counts to integer mode
Warning in DESeqDataSet(se, design = design, ignoreRank) :
  some variables in design formula are characters, converting to factors</code></pre>




<pre class="r"><code>dds</code></pre>


<pre><code>class: DESeqDataSet 
dim: 19084 140 
metadata(1): version
assays(1): counts
rownames(19084): A1BG A1CF ... ZZEF1 ZZZ3
rowData names(0):
colnames(140): d2b10a d2b10b ... d2t9a d2t9b
colData names(1): condition</code></pre>






<pre class="r"><code>#Remove genes with 10 or less reads
genes_to_keep &lt;- rowSums(counts(dds)) &gt;= 10
dds &lt;- dds[genes_to_keep,]</code></pre>






<pre class="r"><code>#run the DESeq wrapper function. 
dds &lt;- DESeq(dds)</code></pre>


<pre><code>estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting for 1085 genes
-- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
-- original counts are preserved in counts(dds)
estimating dispersions
fitting model and testing</code></pre>


<pre class="r"><code>class(dds)</code></pre>


<pre><code>[1] &quot;DESeqDataSet&quot;
attr(,&quot;package&quot;)
[1] &quot;DESeq2&quot;</code></pre>






<pre class="r"><code>#Obtain normalized data from Deseq object raw counts
rawcounts.matrix &lt;- counts(dds,normalized=F)
normalizedcounts.matrix &lt;- counts(dds,normalized=T)</code></pre>






<pre class="r"><code># “rlog” transform the normalized counts and perform hierarchical clustering of samples:
vst_transform &lt;- vst(dds)
dists &lt;- dist(t(assay(vst_transform)))
</code></pre>






<pre class="r"><code>H_Clustering&lt;-(hclust(dists))
par(cex=0.3, mar=c(5, 8, 4, 1))
plot(H_Clustering, xlab=&quot;&quot;, ylab=&quot;&quot;, main=&quot;&quot;, sub=&quot;&quot;, axes=FALSE)
par(cex=1)</code></pre>


<pre class="r"><code>title(xlab=&quot;xlab&quot;, ylab=&quot;ylab&quot;, main=&quot;main&quot;)
axis(2)</code></pre>


<p><img src="javascript://"/></p>



<p>Data clustering exploration with the dendextend package</p>



<pre class="r"><code># Create a dend:
dend &lt;- dists %&gt;% hclust %&gt;% as.dendrogram</code></pre>






<pre class="r"><code>
dend_test &lt;-dists %&gt;% hclust(method = &quot;average&quot;) %&gt;%  as.dendrogram %&gt;% 
   set(&quot;branches_lty&quot;, c(1,1,3,1,1,2)) %&gt;%
   set(&quot;nodes_pch&quot;, 19) %&gt;% set(&quot;nodes_col&quot;, c(&quot;pink&quot;, &quot;purple&quot;, &quot;green&quot;, NA))
# plot the dend in usual &quot;base&quot; plotting engine:
par(cex=0.8)
plot(dend_test)
title(xlab=&quot;Sample Names&quot;, ylab=&quot;Distance&quot;, main=&quot;Hierarchical Clustering of Samples Exposed to toxic and non-toxic chemicals&quot;)
</code></pre>


<p><img src="javascript://"/></p>






<pre class="r"><code>library(circlize)

dend &lt;- dists %&gt;% hclust %&gt;% as.dendrogram %&gt;%
   set(&quot;branches_k_color&quot;, k=3) %&gt;% set(&quot;branches_lwd&quot;, c(5,2,1.5)) %&gt;%
   set(&quot;branches_lty&quot;, c(1,1,3,1,1,2)) %&gt;%
   set(&quot;nodes_pch&quot;, 19) %&gt;% set(&quot;nodes_col&quot;, c(&quot;orange&quot;, &quot;black&quot;, &quot;plum&quot;, NA))

par(mar = rep(0,4))
circlize_dendrogram(dend)</code></pre>


<p><img src="javascript://"/></p>




<pre class="r"><code>pcaData&lt;-plotPCA(vst_transform)
pcaData+geom_label(aes(label = name))+ggtitle(&#39;Principal Component Analysis of RNA Seq Data&#39; )</code></pre>


<p><img src="javascript://"/></p>




<pre class="r"><code>res &lt;- results(dds, contrast = c(&#39;condition&#39;,&#39;Nontoxic&#39;,&#39;Toxic&#39;), alpha = 0.05 )</code></pre>






<pre class="r"><code>plotMA(res, ylim=c(-2,2))</code></pre>


<p><img src="javascript://"/></p>




<pre class="r"><code>diffexpgenes&lt;-as.data.frame(subset(res, res[,5] &lt;0.05  &amp; res[,2] &lt; -1 | res[,2] &gt; 1  ))
sprintf(&#39; The number of significantly expressed genes (p&lt;0.05) and with a log ratio with absolute value above 1 is %s&#39;, nrow(diffexpgenes))</code></pre>


<pre><code>[1] &quot; The number of significantly expressed genes (p&lt;0.05) and with a log ratio with absolute value above 1 is 118&quot;</code></pre>




<pre class="r"><code>resOrdered &lt;- res[order(res$pvalue),] 
head(resOrdered,30) # View the 10 most significant genes</code></pre>


<pre><code>log2 fold change (MLE): condition Nontoxic vs Toxic 
Wald test p-value: condition Nontoxic vs Toxic 
DataFrame with 30 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
MT1X      18.6273       -2.93194  0.312200  -9.39122 5.93092e-21 9.40170e-17
MT2A     111.9069       -2.23805  0.254707  -8.78676 1.53938e-18 1.22011e-14
MT1E      14.0895       -4.55961  0.601770  -7.57701 3.53618e-14 1.86852e-10
GDF15     18.9009       -1.43641  0.222463  -6.45688 1.06883e-10 3.88236e-07
NBL1      14.1159       -1.50216  0.233390  -6.43626 1.22456e-10 3.88236e-07
...           ...            ...       ...       ...         ...         ...
HSPA1A   489.7599      -0.507784 0.1059145  -4.79428 1.63259e-06  0.00099538
SEMA3E    36.6360      -0.639591 0.1342760  -4.76326 1.90492e-06  0.00111840
RARRES3   13.6169      -0.582819 0.1226590  -4.75154 2.01875e-06  0.00114290
HSPA1B   726.8664      -0.398059 0.0844874  -4.71147 2.45941e-06  0.00134436
KLF5      22.7120      -0.457059 0.0975948  -4.68324 2.82380e-06  0.00149210</code></pre>




<pre class="r"><code>norm_counts_top_30=normalizedcounts.matrix[rownames(head(resOrdered,30)), ]</code></pre>






<pre class="r"><code>head(norm_counts_top_30)</code></pre>






<pre class="r"><code>#Create a new matrix or dataframe that contains the expression values from normvalues for just the diffexpgenes and call it diffexpvalues.
diffexp_ID&lt;-rownames(diffexpgenes)
diffexpvalues&lt;-normalizedcounts.matrix[(diffexp_ID),]</code></pre>






<pre class="r"><code>diffexpvalues_corr&lt;-cor((diffexpvalues), method=&#39;pearson&#39;)
diffexpvalues_dist&lt;-as.dist(1-diffexpvalues_corr)
diffexpvalues_clustering &lt;- hclust(dist(diffexpvalues), method = &quot;average&quot;)</code></pre>






<pre class="r"><code>dend_test_diff_exp &lt;-diffexpvalues_dist %&gt;% hclust(method = &quot;average&quot;) %&gt;%  as.dendrogram %&gt;% 
   set(&quot;branches_lty&quot;, c(1,1,3,1,1,2)) %&gt;%
   set(&quot;nodes_pch&quot;, 19) %&gt;% set(&quot;nodes_col&quot;, c(&quot;pink&quot;, &quot;purple&quot;, &quot;green&quot;, NA))
# plot the dend in usual &quot;base&quot; plotting engine:
par(cex=0.8)
plot(dend_test_diff_exp )
title(xlab=&quot;Sample Names&quot;, ylab=&quot;Distance&quot;, main=&quot;Hierarchical Clustering of Samples Exposed to toxic and non-toxic chemicals based on Differentially Expressed Genes&quot;)
</code></pre>


<p><img src="javascript://"/></p>




<pre class="r"><code>diffexpvalues_corr_genes&lt;-cor(t(diffexpvalues), method=&#39;pearson&#39;)
diffexpvalues_dist_genes&lt;-as.dist(1-diffexpvalues_corr_genes)</code></pre>






<pre class="r"><code>dend_test_diff_exp_genes &lt;-diffexpvalues_dist_genes %&gt;% hclust(method = &quot;average&quot;) %&gt;%  as.dendrogram %&gt;% 
   set(&quot;branches_lty&quot;, c(1,1,3,1,1,2)) %&gt;%
   set(&quot;nodes_pch&quot;, 19) %&gt;% set(&quot;nodes_col&quot;, c(&quot;pink&quot;, &quot;purple&quot;, &quot;green&quot;, NA))
# plot the dend in usual &quot;base&quot; plotting engine:
par(cex=0.8)
plot(dend_test_diff_exp_genes )
title(xlab=&quot;Genes Names&quot;, ylab=&quot;Distance&quot;, main=&quot;Hierarchical Clustering of Differentially Expressed Genes&quot;)
</code></pre>


<p><img src="javascript://"/></p>




<pre class="r"><code>sampleCondition &lt;- c(rep(&#39;Nontoxic&#39;,72),rep(&#39;Toxic&#39;,68))
sampleTable &lt;- data.frame(sampleName = colnames(norm_counts_top_30),
                          condition = sampleCondition)</code></pre>






<pre class="r"><code>sampleTable</code></pre>


<div>

</div>






<pre class="r"><code>annotation_columns=sampleTable</code></pre>






<pre class="r"><code>row.names(annotation_columns) &lt;- colnames(norm_counts_top_30)</code></pre>






<pre class="r"><code>annotation_columns$sampleName&lt;-NULL
annotation_columns</code></pre>


<div>

</div>






<pre class="r"><code>pheatmap(norm_counts_top_30, color=colorRampPalette(c(&quot;white&quot;, &quot;light pink&quot;, &quot;purple&quot;))(30), scale=&quot;row&quot;, cluster_cols = T, show_rownames = T,fontsize_row = 4, fontsize_col = 4,,labels_row = rownames(dists),main=&#39;Differentially Expressed genes with hierarchical clustering&#39;,annotation_col =annotation_columns )</code></pre>


<p><img src="javascript://"/></p>



</div>
<div class="section level3">
<h3>GO Term Enrichment</h3>



<pre class="r"><code>library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)</code></pre>




<pre class="r"><code>install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;stephenturner/annotables&quot;)</code></pre>






<pre class="r"><code>library(annotables)</code></pre>






<pre class="r"><code>gene&lt;-rownames(diffexpgenes)</code></pre>




<pre class="r"><code>gene</code></pre>


<pre><code>  [1] &quot;ABCA4&quot;      &quot;AFP&quot;        &quot;ALDH1A2&quot;    &quot;ALDH3A1&quot;    &quot;ALOXE3&quot;     &quot;APOBEC3B&quot;  
  [7] &quot;APOL3&quot;      &quot;ARHGEF15&quot;   &quot;ARSI&quot;       &quot;ASB2&quot;       &quot;BATF2&quot;      &quot;BTLA&quot;      
 [13] &quot;BTN1A1&quot;     &quot;C10orf90&quot;   &quot;C11orf91&quot;   &quot;C19orf38&quot;   &quot;C3orf45&quot;    &quot;C9&quot;        
 [19] &quot;C9orf135&quot;   &quot;CASQ2&quot;      &quot;CCIN&quot;       &quot;CCR7&quot;       &quot;CD300LB&quot;    &quot;CHI3L2&quot;    
 [25] &quot;CKM&quot;        &quot;CLCA2&quot;      &quot;CLDN11&quot;     &quot;CLEC17A&quot;    &quot;CLPS&quot;       &quot;CRB3&quot;      
 [31] &quot;CRYAA&quot;      &quot;CRYBA4&quot;     &quot;CRYBB1&quot;     &quot;CXCL10&quot;     &quot;DDO&quot;        &quot;DGKG&quot;      
 [37] &quot;DLX2&quot;       &quot;DNAH17&quot;     &quot;DNAH3&quot;      &quot;EBI3&quot;       &quot;ELF3&quot;       &quot;FABP4&quot;     
 [43] &quot;FAIM3&quot;      &quot;FAM124B&quot;    &quot;FAM209B&quot;    &quot;FAM71F1&quot;    &quot;FAM83G&quot;     &quot;FITM1&quot;     
 [49] &quot;FOSB&quot;       &quot;FOSL1&quot;      &quot;FOXD4L1&quot;    &quot;FOXD4L6&quot;    &quot;FSD2&quot;       &quot;GALR2&quot;     
 [55] &quot;GDF15&quot;      &quot;GLDN&quot;       &quot;HABP2&quot;      &quot;HHIPL2&quot;     &quot;HIST1H1E&quot;   &quot;HIST1H3E&quot;  
 [61] &quot;HOXA4&quot;      &quot;HOXC4&quot;      &quot;HSPA6&quot;      &quot;IBSP&quot;       &quot;IGFN1&quot;      &quot;IP6K3&quot;     
 [67] &quot;IRS4&quot;       &quot;ITIH1&quot;      &quot;KCNK5&quot;      &quot;KLHL33&quot;     &quot;LOC283710&quot;  &quot;LRRC8E&quot;    
 [73] &quot;LY6K&quot;       &quot;MAFA&quot;       &quot;MCART2&quot;     &quot;MMP10&quot;      &quot;MT1E&quot;       &quot;MT1F&quot;      
 [79] &quot;MT1X&quot;       &quot;MT2A&quot;       &quot;MX2&quot;        &quot;MYBPC2&quot;     &quot;MYEOV&quot;      &quot;MYH8&quot;      
 [85] &quot;MYO1A&quot;      &quot;NBL1&quot;       &quot;NKX3-1&quot;     &quot;NLRP9&quot;      &quot;NR0B1&quot;      &quot;NRAP&quot;      
 [91] &quot;OLR1&quot;       &quot;OR1F1&quot;      &quot;PAX8&quot;       &quot;PCK1&quot;       &quot;PLIN4&quot;      &quot;PP2D1&quot;     
 [97] &quot;RAD21L1&quot;    &quot;RAMP1&quot;      &quot;RFPL4B&quot;     &quot;SERPINA3&quot;   &quot;SERPINE3&quot;   &quot;SLAMF9&quot;    
[103] &quot;SLC25A31&quot;   &quot;SLC2A9&quot;     &quot;SLC6A13&quot;    &quot;SLCO2A1&quot;    &quot;ST20-MTHFS&quot; &quot;ST8SIA6&quot;   
[109] &quot;TDO2&quot;       &quot;TM4SF19&quot;    &quot;TMPRSS11F&quot;  &quot;TMPRSS9&quot;    &quot;TNFRSF8&quot;    &quot;TPO&quot;       
[115] &quot;TSPYL6&quot;     &quot;UBTFL1&quot;     &quot;VGLL2&quot;      &quot;VIPR1&quot;     </code></pre>



</div>
<div class="section level3">
<h3>Export dataframe of differentially expressed genes to csv file.</h3>
<p>Filter the matrix of normalized gene counts using the gene names of the 118 differentially expressed genes (filtered by log fold change and pvalue)</p>



<pre class="r"><code>nrow(diffexpgenes)</code></pre>


<pre><code>[1] 118</code></pre>




<pre class="r"><code>head(diffexpgenes)</code></pre>


<div>

</div>




<pre class="r"><code>rows_to_select&lt;-rownames(diffexpgenes)</code></pre>






<pre class="r"><code>norm_counts_diffexpgenes&lt;-normalizedcounts.matrix[rows_to_select, ]</code></pre>




<pre class="r"><code>nrow(norm_counts_diffexpgenes)</code></pre>


<pre><code>[1] 118</code></pre>




<pre class="r"><code>norm_counts_diffexpgenes&lt;-normalizedcounts.matrix[rows_to_select,]</code></pre>




<pre class="r"><code>length(norm_counts_diffexpgenes)</code></pre>


<pre><code>[1] 16520</code></pre>






<pre class="r"><code>write.csv(norm_counts_diffexpgenes,&#39;/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/RSEM_diff_exp_genes.csv&#39;, row.names = TRUE)</code></pre>






<pre class="r"><code>hs &lt;- org.Hs.eg.db
my.symbols &lt;- gene
GENEID_ENTREZ&lt;-select(hs, 
       keys = my.symbols,
       columns = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;),
       keytype = &quot;SYMBOL&quot;)</code></pre>


<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>


<pre class="r"><code>GENEID_ENTREZ</code></pre>


<div>

</div>






<pre class="r"><code>#Convert the names of all genes tested in the experiment to entrez id
all_gene&lt;-rownames(RSEM_avg)
my.symbols_2 &lt;- all_gene
GENEID_ENTREZ_2&lt;-select(hs, 
       keys = my.symbols_2 ,
       columns = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;),
       keytype = &quot;SYMBOL&quot;)</code></pre>


<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>


<pre class="r"><code>GENEID_ENTREZ_2</code></pre>


<div>

</div>






<pre class="r"><code>ggo &lt;- groupGO(gene     = GENEID_ENTREZ$ENTREZID,
               OrgDb    = org.Hs.eg.db,
               ont      = &quot;BP&quot;,
               level    = 5,
               readable = TRUE)
</code></pre>






<pre class="r"><code>head(ggo)</code></pre>


<div>

</div>






<pre class="r"><code>ego &lt;- enrichGO(gene          = GENEID_ENTREZ$ENTREZID,
                universe      = GENEID_ENTREZ_2$ENTREZID,
                OrgDb         = org.Hs.eg.db,
                ont           = &quot;ALL&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,)
head(ego)</code></pre>


<div>

</div>




</div>

<div>---
title: "RSEM Raw Data DESEQ2 Neurotoxicity Study"
output: html_notebook
---

The raw data for this study was kindly provided by the authors of the paper:

Schwartz, M. P., Hou, Z., Propson, N. E., Zhang, J., Engstrom, C. J., Costa, V. S., Jiang, P., Nguyen, B. K., Bolin, J. M., Daly, W., Wang, Y., Stewart, R., Page, C. D., Murphy, W. L., & Thomson, J. A. (2015). Human pluripotent stem cell-derived neural constructs for predicting neural toxicity. Proceedings of the National Academy of Sciences, 112(40), 12516–12521. https://doi.org/10.1073/pnas.1516645112 

The data is Expected Count data processed with RSEM.

```{r}
install.packages('dendextend')
```
```{r}
install.packages("circlize")
```


```{r}
library(dplyr)
library(pheatmap)
library(DESeq2)
```
```{r}
library(dendextend)
```
```{r}
library(ggplot2)
```


```{r}
RSEM_data=read.csv('/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/RSEM_data.csv')
```

The RSEM data is made up of the data 2 and 7 days after chemical exposure.
```{r}
head(RSEM_data)
```
```{r}
RSEM_day2<-RSEM_data[,1:146]
```
```{r}
RSEM_day2
```


```{r}
#Obtain data from day 2
library(plyr)
RSEM_day2 <- plyr::rename(RSEM_day2, c("d2t11a_reprep"="d2t11a"))
```

```{r}
RSEM_day2 <- plyr::rename(RSEM_day2, c("d2t13a_reprep"="d2t13a"))
```

```{r}
RSEM_day2<-RSEM_day2[,order(colnames(RSEM_day2))]
RSEM_day2$Transcript_ID<-NULL
RSEM_day2
```


```{r}
#Obtain data from day 7
RSEM_day7<-RSEM_data[,147:290]
RSEM_day7<-RSEM_day7[,order(colnames(RSEM_day7))]
RSEM_day7['Gene.ID']<-RSEM_day2$Gene.ID
head(RSEM_day7)


```

Making sure the columns correspond to one another:
```{r}
colnames(RSEM_day7)
```
```{r}
colnames(RSEM_day2)
```
The datafrmes are now sorted and now they can be added in order to obtain a dataframe that represents the average gene expression from day 2 and day 7:
```{r}
RSEM_avg<-(RSEM_day2[,1:144]+RSEM_day7[,1:144])/2
```

The dataframe below can now be used for exploratory data analysis and differential gene expression analysis. 
```{r}
head(RSEM_avg)
```
```{r}
#Remove last four columns labeled 'un'
RSEM_avg$d2un1a=NULL
RSEM_avg$d2un1b=NULL
RSEM_avg$d2un2a=NULL
RSEM_avg$d2un2b=NULL
```

```{r}
#set rownames to gene id
rownames(RSEM_avg)<-RSEM_data$Gene.ID
```

```{r}
head(RSEM_avg)
```


Save Dataframe to csv file:
```{r}
write.csv(RSEM_avg,'/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/RSEM_avg.csv', row.names = FALSE)
```

### Differential Gene Expression Analysis
```{r}
expgroup<-data.frame()
condition_toxic<-rep('Toxic',68,)
condition_nontoxic<-rep('Nontoxic',72)
condition<-c(condition_nontoxic,condition_toxic)
samples<-colnames(RSEM_avg)
expgroup<-cbind(condition)
rownames(expgroup)<-samples
```

```{r}
expgroup
```
```{r}
#Round values in count matrix for DESEQ2 input
RSEM_avg<-round(RSEM_avg,0)
```


```{r}
dds <-DESeqDataSetFromMatrix(
  countData=RSEM_avg,
  colData=expgroup,
  design=~condition,
)
```
```{r}
dds
```


```{r}
#Remove genes with 10 or less reads
genes_to_keep <- rowSums(counts(dds)) >= 10
dds <- dds[genes_to_keep,]
```

```{r}
#run the DESeq wrapper function. 
dds <- DESeq(dds)
class(dds)
```

```{r}
#Obtain normalized data from Deseq object raw counts
rawcounts.matrix <- counts(dds,normalized=F)
normalizedcounts.matrix <- counts(dds,normalized=T)
```

```{r}
# “rlog” transform the normalized counts and perform hierarchical clustering of samples:
vst_transform <- vst(dds)
dists <- dist(t(assay(vst_transform)))

```

```{r out.width="100%"}
H_Clustering<-(hclust(dists))
par(cex=0.3, mar=c(5, 8, 4, 1))
plot(H_Clustering, xlab="", ylab="", main="", sub="", axes=FALSE)
par(cex=1)
title(xlab="xlab", ylab="ylab", main="main")
axis(2)
```

Data clustering exploration with the dendextend package
```{r}
# Create a dend:
dend <- dists %>% hclust %>% as.dendrogram
```


```{r}

dend_test <-dists %>% hclust(method = "average") %>%  as.dendrogram %>% 
   set("branches_lty", c(1,1,3,1,1,2)) %>%
   set("nodes_pch", 19) %>% set("nodes_col", c("pink", "purple", "green", NA))
# plot the dend in usual "base" plotting engine:
par(cex=0.8)
plot(dend_test)
title(xlab="Sample Names", ylab="Distance", main="Hierarchical Clustering of Samples Exposed to toxic and non-toxic chemicals")

```


```{r}
library(circlize)

dend <- dists %>% hclust %>% as.dendrogram %>%
   set("branches_k_color", k=3) %>% set("branches_lwd", c(5,2,1.5)) %>%
   set("branches_lty", c(1,1,3,1,1,2)) %>%
   set("nodes_pch", 19) %>% set("nodes_col", c("orange", "black", "plum", NA))

par(mar = rep(0,4))
circlize_dendrogram(dend)
```
```{r}
pcaData<-plotPCA(vst_transform)
pcaData+geom_label(aes(label = name))+ggtitle('Principal Component Analysis of RNA Seq Data' )
```
```{r}
res <- results(dds, contrast = c('condition','Nontoxic','Toxic'), alpha = 0.05 )
```

```{r}
plotMA(res, ylim=c(-2,2))
```
```{r}
diffexpgenes<-as.data.frame(subset(res, res[,5] <0.05  & res[,2] < -1 | res[,2] > 1  ))
sprintf(' The number of significantly expressed genes (p<0.05) and with a log ratio with absolute value above 1 is %s', nrow(diffexpgenes))
```
```{r}
resOrdered <- res[order(res$pvalue),] 
head(resOrdered,30) # View the 10 most significant genes
```
```{r}
norm_counts_top_30=normalizedcounts.matrix[rownames(head(resOrdered,30)), ]
```

```{r}
head(norm_counts_top_30)
```

```{r}
#Create a new matrix or dataframe that contains the expression values from normvalues for just the diffexpgenes and call it diffexpvalues.
diffexp_ID<-rownames(diffexpgenes)
diffexpvalues<-normalizedcounts.matrix[(diffexp_ID),]
```

```{r}

diffexpvalues_corr<-cor((diffexpvalues), method='pearson')
diffexpvalues_dist<-dist(1-diffexpvalues_corr)
```

```{r}
dend_test_diff_exp <-diffexpvalues_dist %>% hclust(method = "average") %>%  as.dendrogram %>% 
   set("branches_lty", c(1,1,3,1,1,2)) %>%
   set("nodes_pch", 19) %>% set("nodes_col", c("pink", "purple", "green", NA))
# plot the dend in usual "base" plotting engine:
par(cex=0.8)
plot(dend_test_diff_exp )
title(xlab="Sample Names", ylab="Distance", main="Hierarchical Clustering of Samples Exposed to toxic and non-toxic chemicals based on Differentially Expressed Genes")

```
```{r}
diffexpvalues_corr_genes<-cor(t(diffexpvalues), method='pearson')
diffexpvalues_dist_genes<-as.dist(1-diffexpvalues_corr_genes)
```

```{r}
dend_test_diff_exp_genes <-diffexpvalues_dist_genes %>% hclust(method = "average") %>%  as.dendrogram %>% 
   set("branches_lty", c(1,1,3,1,1,2)) %>%
   set("nodes_pch", 19) %>% set("nodes_col", c("pink", "purple", "green", NA))
# plot the dend in usual "base" plotting engine:
par(cex=0.8)
plot(dend_test_diff_exp_genes )
title(xlab="Genes Names", ylab="Distance", main="Hierarchical Clustering of Differentially Expressed Genes")

```
```{r}
sampleCondition <- c(rep('Nontoxic',72),rep('Toxic',68))
sampleTable <- data.frame(sampleName = colnames(norm_counts_top_30),
                          condition = sampleCondition)
```

```{r}
sampleTable
```

```{r}
annotation_columns=sampleTable
```

```{r}
row.names(annotation_columns) <- colnames(norm_counts_top_30)
```


```{r}
annotation_columns$sampleName<-NULL
annotation_columns
```



```{r}
pheatmap(norm_counts_top_30, color=colorRampPalette(c("white", "light pink", "purple"))(30), scale="row", cluster_cols = T, show_rownames = T,fontsize_row = 4, fontsize_col = 4,,labels_row = rownames(dists),main='Differentially Expressed genes with hierarchical clustering',annotation_col =annotation_columns )
```

### GO Term Enrichment

```{r}
library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)
```
```{r}
install.packages("devtools")
devtools::install_github("stephenturner/annotables")
```

```{r}
library(annotables)
```

```{r}
gene<-rownames(diffexpgenes)
```
```{r}
gene
```
### Export dataframe of differentially expressed genes to csv file. 

Filter the matrix of normalized gene counts using the gene names of the 118 differentially expressed genes (filtered  by log fold change and pvalue)

```{r}
nrow(diffexpgenes)
```
```{r}
head(diffexpgenes)
```
```{r}
rows_to_select<-rownames(diffexpgenes)
```

```{r}
norm_counts_diffexpgenes<-normalizedcounts.matrix[rows_to_select, ]
```
```{r}
nrow(norm_counts_diffexpgenes)
```
```{r}
norm_counts_diffexpgenes<-normalizedcounts.matrix[rows_to_select,]
```
```{r}
length(norm_counts_diffexpgenes)
```


```{r}
write.csv(norm_counts_diffexpgenes,'/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/RSEM_diff_exp_genes.csv', row.names = TRUE)
```


```{r}
hs <- org.Hs.eg.db
my.symbols <- gene
GENEID_ENTREZ<-select(hs, 
       keys = my.symbols,
       columns = c("ENTREZID", "SYMBOL"),
       keytype = "SYMBOL")
GENEID_ENTREZ
```

```{r}
#Convert the names of all genes tested in the experiment to entrez id
all_gene<-rownames(RSEM_avg)
my.symbols_2 <- all_gene
GENEID_ENTREZ_2<-select(hs, 
       keys = my.symbols_2 ,
       columns = c("ENTREZID", "SYMBOL"),
       keytype = "SYMBOL")
GENEID_ENTREZ_2
```


```{r}
ggo <- groupGO(gene     = GENEID_ENTREZ$ENTREZID,
               OrgDb    = org.Hs.eg.db,
               ont      = "BP",
               level    = 5,
               readable = TRUE)

```

```{r}
head(ggo)
```

```{r}
ego <- enrichGO(gene          = GENEID_ENTREZ$ENTREZID,
                universe      = GENEID_ENTREZ_2$ENTREZID,
                OrgDb         = org.Hs.eg.db,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,)
head(ego)
```





</div>



</div>
















<script type="module" src="https://s.brightspace.com/lib/bsi/20.22.5-185/unbundled/mathjax.js"></script><script type="text/javascript">document.addEventListener('DOMContentLoaded', function() {
					if (document.querySelector('math') || /\$\$|\\\(|\\\[|\\begin{|\\ref{|\\eqref{/.test(document.body.innerHTML)) {
						document.querySelectorAll('mspace[linebreak="newline"]').forEach(elm => {
							elm.setAttribute('style', 'display: block; height: 0.5rem;');
						});

						window.D2L.MathJax.loadMathJax({
							'outputScale': 1.3,
							'renderLatex': true
						});
					}
				});</script></body></html>